### PR TITLE
HAI-2749 Split UpdateHakemusITest from HakemusServiceITest

### PR DESCRIPTION
**Pull request by corvidian**
# Description

The update tests are half of the hakemus service tests, and it's a really long file. The update tests can be split into a separate file.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [X] Other

### CODE REVIEW

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt [hunk 1]
```diff
@@ -41,36 +39,21 @@ import fi.hel.haitaton.hanke.allu.Contact
...
 import fi.hel.haitaton.hanke.factory.ApplicationFactory.Companion.DEFAULT_EXCAVATION_NOTIFICATION_IDENTIFIER
-import fi.hel.haitaton.hanke.factory.ApplicationFactory.Companion.createExcavationNotificationArea
-import fi.hel.haitaton.hanke.factory.ApplicationFactory.Companion.createTyoalue
 import fi.hel.haitaton.hanke.factory.CreateHakemusRequestFactory
 import fi.hel.haitaton.hanke.factory.DateFactory
 import fi.hel.haitaton.hanke.factory.GeometriaFactory
 import fi.hel.haitaton.hanke.factory.HakemusFactory
-import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory
-import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.toUpdateRequest
 import fi.hel.haitaton.hanke.factory.HakemusUpdateRequestFactory.withArea
```
Comment (umeetiusbaar) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt:56]: This import seems to be unnecessary in this class.

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/UpdateHakemusITest.kt [hunk 2]
```diff
@@ -0,0 +1,1039 @@
...
+    @Autowired private val hakemusFactory: HakemusFactory,
+    @Autowired private val hankeFactory: HankeFactory,
+    @Autowired private val hankeKayttajaFactory: HankeKayttajaFactory,
+    @Autowired private val alluClient: AlluClient,
+) : IntegrationTest() {
+
+    companion object {
+        @JvmField
+        @RegisterExtension
+        val greenMail: GreenMailExtension =
```
Comment (umeetiusbaar) [inline services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/UpdateHakemusITest.kt:91]: Is this needed in this test class?

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt (hunk 1 above)
Comment (corvidian) [reply services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/HakemusServiceITest.kt]: Yes, thank you. Removed.

File: services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/UpdateHakemusITest.kt (hunk 2 above)
Comment (corvidian) [reply services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/hakemus/UpdateHakemusITest.kt:91]: It's necessary here. It's used to test the notification emails when a käyttäjä is added to a hakemus.

It's not used in `HakemusServiceITest.kt`, so removing it from there.

**Review by umeetiusbaar — APPROVED** [review]

**Pull request merged**